File: backend/src/infrastructure/database/models/project.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, Text, DateTime
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from src.infrastructure.database.session import Base

class ProjectModel(Base):
    __tablename__ = "projects"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    location = Column(String(255), nullable=True)
    base_year = Column(Integer, nullable=False)
    planning_horizon = Column(Integer, nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)

    # Relationships
    scenarios = relationship("ScenarioModel", back_populates="project", cascade="all, delete-orphan")
    tenants = relationship("TenantModel", back_populates="project", cascade="all, delete-orphan")
    study_port = relationship("StudyPortModel", back_populates="project", uselist=False, cascade="all, delete-orphan")

File: frontend/src/types/project.ts
import { StudyPort, StudyPortCreate } from "./studyPort";

export interface Project {
  id: string;
  name: string;
  description?: string | null;
  location?: string | null;
  base_year: number;
  planning_horizon: number;
  status: string;
  created_at: string;
  updated_at: string;
  study_port?: StudyPort | null;
}

export interface CreateProjectPayload {
  name: string;
  description?: string;
  location?: string;
  base_year: number;
  planning_horizon: number;
  study_port?: StudyPortCreate;
  copy_study_port_from_project_id?: string;
}

export interface ProjectParameter {
  id: string;
  project_id: string;
  parameter_key: string;
  parameter_value: number;
  created_at: string;
  updated_at: string;
}
